api/main_endpoints/routes/User.js

```diff
@@ -199,6 +207,73 @@ router.post('/getPagesPrintedCount', (req, res) => {
   });
 });
 
+// Verifies the users session if they have an active jwtToken.
+// Used on the inital load of root '/'
+// Returns the name and accesslevel of the user w/ the given access token
+router.post('/verify', function(req, res) {
+  if (!checkIfTokenSent(req)) {
+    return res.sendStatus(UNAUTHORIZED);
+  }
+
+  const decoded = checkIfTokenValid(req);
+  if (decoded) {
+    res.status(OK).send(decoded);
+  } else {
+    res.sendStatus(UNAUTHORIZED);
+  }
+});
+
+router.get('/callback', async function(req, res) {
+  const code = req.query.code;
+  const email = req.query.state;
+  axios
+    .post('https://discord.com/api/oauth2/token',
+      `client_id=${discordApiKeys.CLIENT_ID}` +
+      `&client_secret=${discordApiKeys.CLIENT_SECRET}` +
+      `&grant_type=client_credentials&code=${code}` +
+      `&redirect_uri=${discordApiKeys.REDIRECT_URI}`)
+    .then(response => {
+      axios.get('https://discord.com/api/users/@me', {
+        headers: {
+          'Authorization': `Bearer ${response.data.access_token}`,
+        }
+      })
+        .then(loginResponse => {
+          User.findOne({email:email})
+            .then((user) => {
+              user.discordID = loginResponse.data.id;
+              user.save();
+              return res.redirect('https://discord.com/oauth2/authorized');
+            })
+            .catch(() => {
```

Comment:
you can keep the catch on line 252 and it will catch any errors within these brackets

Reply:
so the one on 248 isn't needed then?